### ISSUE TITLE
Basic question: where is the code that interacts with Nodejs workers?

### ISSUE THREAD

**Issue by alshdavid** [evanw/esbuild#3616]
Apologies for the basic question, I am trying to learn more about how esbuild interact with Node.js when used to interpret JavaScript plugins.

 I've been reading through the source and have found the code that contains the Node.js side to interpret messages [stdio_protocol.ts](https://github.com/evanw/esbuild/blob/main/lib/shared/stdio_protocol.ts).

But I don't see where esbuild is initializing Node.js processes to talk to esbuild from.

My assumption is that esbuild create Node.js children from Go and communicates with them via stdio but I can't find any instances of `exec.Command` or within the Go code.

I do see [some logic](https://github.com/evanw/esbuild/blob/2af5ccf478812d2d7226ad4435d46fbbb3419a8c/lib/npm/node.ts#L257) that seems to probe for a running esbuild process? 

Does Esbuild use a Node.js entry that spawns the Go side as a child?

**Comment by hyrious** [comment]
> Does Esbuild use a Node.js entry that spawns the Go side as a child?

Yes. It does work in this way when you're using its JavaScript API.

On the other hand, when you call `npx esbuild` or more directly `node_modules/.bin/esbuild`, it points to the real executable binary file of esbuild, which does not involve a Node.js runtime.
